hw/ip/i2c/doc/_index.md

```diff
@@ -142,6 +143,30 @@ Typically every byte transmitted must also receive an ACK signal, and the IP wil
 However, there are some I2C commands which do not require an ACK.
 In those cases this flag should be asserted with FBYTE indicating no ACK is expected and no interrupt should be raised if the ACK is not received. 
 
+### Target Address Registers
+
+I2C target device is assigned two 7-bit address and 7-bit mask pairs.
+The target device accepts a transaction if the result of the bitwise AND operation performed on the transaction address sent by the host and a mask matches the assigned address corresponding to the mask.
+Thus, the target device may respond to any of the two assigned unique addresses.
+The assigned address and mask pairs are set in registers {{< regref ADDRESS0 >}}, {{< regref MASK0 >}}, {{< regref ADDRESS1 >}}, and {{< regref MASK1 >}}. 
+
+### Acquired Formatted Data
+
+This section applies to I2C in the target mode.
+When the target accepts a transaction, it inserts the transaction address, read/write bit, and START signal sent by the host into the ACQ FIFO where they can be accessed by software.
+ACQ FIFO output corresponds to {{< regref ACQDATA >}}.
+If the transaction is a write operation (R/W bit = 0), the target proceeds to read bytes from the bus and insert them into the ACQ FIFO until the host terminates the transaction by sending a STOP or RESTART signal.
+A STOP or RESTART indicator is inserted into the ACQ FIFO as the next entry following the last byte read, in which case other bits are junk.
```

Comment:
But we may also encounter a case when the host decides to transact with another device. It will issue a RESTART followed by a different address. Fo the first target, a RESTART will be inserted into ACQ FIFO as a STOP indicator+junk.

If the address that the host sends after the RESTART matches the address of the first target, then the next entry will be START indicator + address + r/w/ bit. How about this?

Reply:
I didn't realize it was legal to do a repeated start with a different address (because logically everything from the first START to a STOP is a single transaction and the bus is busy the whole time). But I don't see anything in the spec (I2C or SMBus) that actually says this so we probably do need to cover this case. 

I don't think we can just fake a stop for the repeated start case (because for some transactions you need them to look like a single transaction rather than thinking the transaction is done eg the preamble to SMBus ReadByte looks just like SMBus SendByte apart from the repeated start vs stop). So I suppose you would put start plus junk and then the start plus address/rw. Or maybe you could set both bits so start+stop with junk = repeated start indication and the next entry will be start+address/rw if it is to this target?

Repeated start with different address seems like a good way for a host on a multi-host bus to steal all the bandwidth!

Reply:
I like indicating a RESTART by setting both start and stop bits to 1 (the remainder fo the entry is junk).